include/aspect/mesh_deformation/fastscape.h

```diff
@@ -0,0 +1,680 @@
+   * hillslope diffusion, the stream power law, sediment deposition
+   * and marine diffusion.
+   */
+  template<int dim>
+  class FastScape : public Interface<dim>, public SimulatorAccess<dim>
+  {
+    public:
+      /**
+       * Initialize variables for FastScape.
+       */
+      virtual void initialize ();
+
+      /**
+        * A function that creates constraints for the velocity of certain mesh
+        * vertices (e.g. the surface vertices) for a specific boundary.
+        * The calling class will respect
+        * these constraints when computing the new vertex positions.
+        */
+      virtual
+      void
+      compute_velocity_constraints_on_boundary(const DoFHandler<dim> &mesh_deformation_dof_handler,
+                                               AffineConstraints<double> &mesh_velocity_constraints,
+                                               const std::set<types::boundary_id> &boundary_id) const;
+
+      /**
+       * Declare parameters for the FastScape plugin.
+       */
+      static
+      void declare_parameters (ParameterHandler &prm);
+
+      /**
+       * Parse parameters for the FastScape plugin.
+       */
+      void parse_parameters (ParameterHandler &prm);
+
+      /**
+       * Function used to set the FastScape ghost nodes. FastScape boundaries are
+       * not uplifted or periodic for advection and diffusion. By using a layer
+       * of extra nodes in the FastScape model, we can avoid seeing FastScape boundary
+       * effects within ASPECT. Similalrly, we can use these nodes to have fully
+       * periodic boundaries, where we check the flow direction and update the FastScape
+       * ghost nodes, and the nodes one layer inward (boundary nodes in ASPECT) to match
+       * the parameters on the other side (vx, vy, vz, h). This is done every ASPECT timestep
+       * before running FastScape.
+       * TODO: Should this be in private?
+       */
+      void set_ghost_nodes(double *h, double *vx, double *vy, double *vz) const;
+
+      /**
+       * Function to fill the Fastscape arrays (height and velocities) with the data received from ASPECT in the correct index order.
+       */
+      void fill_fastscape_arrays(double *h, double *kd, double *kf, double *vx, double *vy, double *vz, std::vector<std::vector<double>> temporary_variables) const;
+
+      /**
+       * Function to get the ASPECT topography and velocities at the surface, and an index for transferring these to FastScape.
+       */
+      std::vector<std::vector<double>> get_aspect_values() const;
+
+      /**
+       * Function to intialize or restart FastScape
+       */
+      void initialize_fastscape(double *h, double *b, double *kd, double *kf, double *sf) const;
+
+      /**
+       * Execute FastScape
+       */
+      void execute_fastscape(double *h, double *kd, int istep) const;
+
+      /**
+       * Function to apply orographic controls to the FastScape model.
```

Comment:
Again as someone not familiar with surface evolution, can you define "orographic" here?